src/shared/ui/Input/lib/interfaces.ts

```diff
@@ -0,0 +1,11 @@
+import { HTMLInputTypeAttribute } from "react"
+
+export interface IInputCommonProps {
+  type: HTMLInputTypeAttribute
+  autoComplete?: string
+
+  name: string
+  placeholder?: string
+  onChange?: (e: string) => void
```

Comment:
Please take into account that you will probably need to have Inputs of type "number" and you will be expecting them to work with numbers in order to pass a form validation process